FindIntelMKL.cmake

```diff
@@ -296,7 +306,11 @@ if( "ilp64" IN_LIST IntelMKL_FIND_COMPONENTS )
   set( IntelMKL_COMPILE_DEFINITIONS "MKL_ILP64" )
   if( CMAKE_C_COMPILER_ID MATCHES "GNU" )
     set( IntelMKL_C_COMPILE_FLAGS        "-m64" )
+  endif()
+  if( CMAKE_Fortran_COMPILER_ID MATCHES "GNU" )
     set( IntelMKL_Fortran_COMPILE_FLAGS  "-m64" "-fdefault-integer-8" )
+  elseif( CMAKE_Fortran_COMPILER_ID MATCHES "Flang" )
+    set( IntelMKL_Fortran_COMPILE_FLAGS  "-fdefault-integer-8" )
   elseif( CMAKE_C_COMPILER_ID MATCHES "PGI" )
     set( IntelMKL_Fortran_COMPILE_FLAGS "-i8" )
   endif()
```

Comment:
We may want to factor this out into a separate `Find` module (or something, not sure of the proper convention here) to handle integer based flags. Logic follows that these should also propagate to other BLAS distros in the case that ILP64 is requested